### PR TITLE
feat: add BatchNormalization layer to TensorMap model builder

### PR DESCRIPTION
**Pull request by Sickwoman**
## Summary
Adds full-stack support for the BatchNormalization layer in TensorMap,
allowing users to add batch normalization between layers in the model builder canvas.

### Backend
- **`app/services/model_generation.py`** — Added `custombatchnorm` node handler:
```python
elif node_type == "custombatchnorm":
    return tf.keras.layers.BatchNormalization(
        momentum=float(params.get("momentum", 0.99)),
        epsilon=float(params.get("epsilon", 0.001)),
        name=name,
    )(input_tensor)
```
- **`tests/test_model_generation.py`** — Added unit tests for BatchNormalization layer

### Frontend
- **`Canvas.jsx`** — Registered `custombatchnorm` node type with default params `{ momentum: "0.99", epsilon: "0.001" }`
- **`Helpers.jsx`** — Added validation for `custombatchnorm` node (momentum and epsilon required)
- **`Sidebar.jsx`** — Added BatchNorm drag-and-drop item to the layer sidebar
- **`BatchNormalizationNode/`** — New custom node component with momentum and epsilon input fields

## Parameters
| Parameter | Default | Description |
|-----------|---------|-------------|
| `momentum` | `0.99` | Momentum for the moving average |
| `epsilon` | `0.001` | Small float added to variance to avoid division by zero |

## How to Test
1. Open the model builder canvas
2. Drag **BatchNorm** from the sidebar into the canvas
3. Connect it between two layers (e.g. Dense → BatchNorm → Dense)
4. Set momentum and epsilon values
5. Validate and train the model

## No Breaking Changes
All existing layer types and model builder functionality are unchanged.

### CODE REVIEW

**Comment by Sickwoman** [comment]
Reopened PR#252 as suggested.

**Comment by ivantha** [comment]
Closing this in line with the TensorMap `gsoc` label convention. Issue #120 reserves the full scope of the GSoC 2026 student's term, and this PR overlaps with both **Phase 1 (Foundation: layer registry refactor)** and **Phase 2 (Layer expansion: 11 new layer types including regularization)**. Adding BatchNormalization with a per-file Canvas/Sidebar/Helpers/NodePropertiesPanel touch is exactly the 6-file pattern that the Phase 1 registry refactor is meant to replace, and BatchNorm itself is one of the regularization layers Phase 2 introduces.

Coding for the selected contributor (Kartikey Gupta) begins on 2026-05-25, so this surface is fenced off until then. Thanks for the contribution; if you would like to contribute before or around GSoC, please pick up a `pre-gsoc` labeled issue instead.
